ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/identity_provider_config_service.proto

```diff
@@ -0,0 +1,141 @@
+// (2) ``Required`` denoting the client must set the field to a non-default value when sending a request.
+//
+// An identity provider config resource, once it has been created, can be modified.
+// In order to update the properties represented by the ``IdentityProviderConfig`` message use the ``UpdateIdentityProviderConfig`` RPC.
+// The only fields that can be modified are those marked as ``Modifiable``.
+service IdentityProviderConfigService {
+
+  rpc CreateIdentityProviderConfig (CreateIdentityProviderConfigRequest) returns (CreateIdentityProviderConfigResponse);
+
+  rpc GetIdentityProviderConfig (GetIdentityProviderConfigRequest) returns (GetIdentityProviderConfigResponse);
+
+  rpc ListIdentityProviderConfigs (ListIdentityProviderConfigsRequest) returns (ListIdentityProviderConfigsResponse);
+
+  rpc UpdateIdentityProviderConfig (UpdateIdentityProviderConfigRequest) returns (UpdateIdentityProviderConfigResponse);
+
+  rpc DeleteIdentityProviderConfig (DeleteIdentityProviderConfigRequest) returns (DeleteIdentityProviderConfigResponse);
+
+}
+
+message IdentityProviderConfig {
+  // The identity provider config identifier, which must be a non-empty string of at most 128
+  // characters that are either lowercase alphanumeric ASCII characters or one of the symbols "@^$.!`-#+'~_|:".
+  // Required
+  string id = 1;
+
+  // When set, then the user identified by the identity provider is denied all access to the Ledger API.
+  // Otherwise, the user identified by the identity provider has access to the Ledger API as per the user's rights.
+  // Optional,
+  // Modifiable
+  bool is_deactivated = 2;
+
+  // Specifies the issuer of the JWT token that should be used in the "iss" claim of a JWT token to instruct the
+  // participant node to use this authentication domain to authenticate the token.
+  // The issuer value is a case sensitive URL using the https scheme that contains scheme, host,
+  // and optionally, port number and path components and no query or fragment components.
+  // Required
+  // Modifiable
+  string issuer = 3;
+
+  // The JWKS (JSON Web Key Set) URL.
+  // The Ledger API will expect all tokens to be signed with RS256 (RSA Signature with SHA-256) with the public
+  // key loaded from the given URL.
+  // Required
+  // Modifiable
+  string jwks_uri = 4;
+}
+
+
+message CreateIdentityProviderConfigRequest {
+  // Required
+  IdentityProviderConfig identity_provider_config = 1;
+}
+
+message CreateIdentityProviderConfigResponse {
+  IdentityProviderConfig identity_provider_config = 1;
+}
+
+message GetIdentityProviderConfigRequest {
+  // Required
+  string identity_provider_config_id = 1;
+}
+
+message GetIdentityProviderConfigResponse {
+  IdentityProviderConfig identity_provider_config = 1;
+}
+
+message ListIdentityProviderConfigsRequest {
+  // Pagination token to determine the specific page to fetch.
+  // Leave empty to fetch the first page.
+  // Optional
+  string page_token = 1;
```

Comment:
If we configure so many IDPs on a participant that we need to paginate them, then we are lost already. I suggest introducing a hard limit of let's say 16 IDPs

Reply:
Ok, I removed the pagination, but this limit I guess we should not be mentioning in the proto, but rather making this limit configurable in the hocon configuration.